src/middlewared/middlewared/api/v25_04_0/api_key.py

```diff
@@ -0,0 +1,62 @@
+from datetime import datetime
+from typing import Literal, TypeAlias
+from typing_extensions import Annotated
+
+from pydantic import ConfigDict, StringConstraints
+
+from middlewared.api.base import BaseModel, Excluded, excluded_field, NonEmptyString, Private
+
+
+HttpVerb: TypeAlias = Literal["GET", "POST", "PUT", "DELETE", "CALL", "SUBSCRIBE", "*"]
+
+
+class AllowListItem(BaseModel):
+    method: HttpVerb
+    resource: NonEmptyString
+
+
+class ApiKeyEntry(BaseModel):
+    """Represents a record in the account.api_key table."""
+    #: This allows the model to be created from an instance of plugins/api_key.APIKeyModel
```

Comment:
Sorry, what does this do, exactly? We don't have instances of `APIKeyModel` anywhere, we don't use SQLAlchemy ORM

Reply:
It just allows you to validate and create an `ApiKeyEntry` from an `APIKeyModel`. I only added this as an option because I noticed it matches the structure of the ORM model we already have. I can remove if it's unnecessary.

Reply:
https://docs.pydantic.dev/latest/concepts/models/#arbitrary-class-instances

Reply:
We don't have any instances of SQLalchemy ORM models and we never will, so this feature won't be used and can be removed.